src/Interpreters/Set.cpp

```diff
@@ -430,8 +430,9 @@ MergeTreeSetIndex::MergeTreeSetIndex(const Columns & set_elements, std::vector<K
     SortDescription sort_description;
     for (size_t i = 0; i < tuple_size; ++i)
     {
-        block_to_sort.insert({ordered_set[i], nullptr, ordered_set[i]->getName()});
-        sort_description.emplace_back(ordered_set[i]->getName(), 1, 1);
+        String column_name = "_" + toString(i);
```

Comment:
LGTM, but I don't understand, what was the problem?

Reply:
`ordered_set[i]` is a `ColumnPtr` and `getName` is not unique obviously. In result we could get wrong sorting of block.

Reply:
Maybe also add an assert to `sortBlock`? Seems that it should not have duplicated columns in `description` they should be filtered out before.

Reply:
I'm not sure that it's a good assertion, because sort description with duplicated columns is a valid sort description and removing them is just an optimization. I'd prefer to add such assertion to `Block`, since duplicated columns break the invariant and make behaviour of `Block::getByName` undefined. But as I remember it's can't be done now, because we can have blocks with duplicated columns during analysis of query.